src/test/java/com/vkatit/cinema/CinemaApplicationTests.java

```diff
@@ -14,9 +28,43 @@ class CinemaApplicationTests {
 	@Autowired
 	ApplicationContext context;
 
+	@Autowired
+	private DownloadController downloadController;
+
+	@Value("${ticket.path}")
+	private String TICKET_PATH;
+
 	@Test
 	void contextLoads() {
 		assertTrue(context != null);
 	}
 
+	@Test
+	@DirtiesContext
+	public void fileDownload() {
+		String fileName = "ticketid.pdf";
+		Path filePath = Paths.get(TICKET_PATH, fileName);
+		try {
+			Resource resource = new FileSystemResource(filePath.toFile());
+			HttpHeaders headers = new HttpHeaders();
+			headers.add(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=" + fileName);
+			headers.setContentType(MediaType.parseMediaType("application/octet-stream"));
+			ResponseEntity<Resource> response = (ResponseEntity<Resource>) downloadController.download(fileName);
```

Comment:
it's unit test. not integration test. integratin test in that case is: start application, make http request, get response from http call.

Reply:
The test has been converted to integration